cmd/crypto/error.go

```diff
@@ -14,60 +14,79 @@
 
 package crypto
 
-import "errors"
+import (
+	"fmt"
+)
 
 // Error is the generic type for any error happening during decrypting
 // an object. It indicates that the object itself or its metadata was
 // modified accidentally or maliciously.
-type Error string
+type Error struct {
+	err error
+}
+
+// Errorf - formats according to a format specifier and returns
+// the string as a value that satisfies error of type crypto.Error
+func Errorf(format string, a ...interface{}) error {
+	return Error{err: fmt.Errorf(format, a...)}
+}
 
-func (e Error) Error() string { return string(e) }
+// Unwrap the internal error.
+func (e Error) Unwrap() error { return e.err }
+
+// Error 'error' compatible method.
+func (e Error) Error() string {
+	if e.err == nil {
+		return "<nil>"
```

Comment:
```suggestion
		return ""
```
If there is no error we should default to the zero value for string which is `""`.

Reply:
The behavior is same as `fmt.Println(err)` if err = nil @aead

Reply:
I don't see an empty string from Go so this behavior is kept in line with that @aead

Reply:
@harshavardhana yes but consider:
- `fmt.Println(error(nil))`
- `fmt.Println(Error{})`
Both will show the same result - however, these are distinct cases. That's one of my reasons against an error type that "just" wraps another error without any other context.

Originally the `Error` type has been introduced to have a `crypto.Error` type for `cmd` type assert (additional to == comparison to known errors) to return an appropriate error code to the client.

Maybe: return `crypto: cause <nil>` instead?